docs/federation/federation-directives-index.mdx

```diff
@@ -0,0 +1,31 @@
+description: "An index of directives supported by WunderGraph Cosmo."
+icon: "list-check"
+---
+
+This page lists the GraphQL Federation directives that WunderGraph Cosmo supports. Each entry includes a brief description. For usage details, see the directive’s documentation or related federation guides.
+
+| Directive                        | Explanation                                                                                                                                        |
+| -------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [`@authenticated`](/federation/directives/authenticated)                 | Marks a field or type as requiring authentication. Only requests with valid credentials can access it.                                             |
+| `@composeDirective` (planned)    | Tells the composition process to preserve a specific custom type system directive in the supergraph.                                               |
+| [`@edfs__kafkaPublish`](/federation/event-driven-federation/kafka)            | Marks a mutation field that publishes an event to a Kafka topic. Requires a topic and provider ID.                                                 |
+| [`@edfs__kafkaSubscribe`](/federation/event-driven-federation/kafka)          | Declares a subscription field that listens to one or more Kafka topics. Used in EDG Subscription fields.                                           |
+| [`@edfs__natsPublish`](/federation/event-driven-federation/nats)             | Marks a mutation field that publishes an event to a NATS subject. Requires a subject and provider ID.                                              |
+| [`@edfs__natsRequest`](/federation/event-driven-federation/nats)             | Declares a query field that requests an event over NATS and expects a response. Used in EDG Query fields.                                          |
+| [`@edfs__natsSubscribe`](/federation/event-driven-federation/nats)           | Marks a subscription field that listens to one or more NATS subjects. Can include stream configuration.                                            |
+| `@external`                      | Indicates that a field (or all fields of a type) comes from another subgraph and isn’t resolved here.                                              |
+| `@extends`                       | Marks an object or interface as an extension of a type defined in another subgraph, when `extend` isn’t used.                                      |
+| `@inaccessible`                  | Hides a field or type from the API schema while keeping it in the supergraph for internal use.                                                     |
+| `@interfaceObject`               | Lets a subgraph contribute fields to an interface and all implementing types by defining an object abstraction.                                    |
```

Comment:
We mark with it a type which is an interface in the other subgraph and this interface has a key directive attached, such interface is called entity interface 

It allows to contribute fields to every type implementing entity interface in the other subgraph 

When you use interface object it is not allowed to have in the same subgraphs any implementing types, otherwise it will be a composition error